numpy/core/include/numpy/utils.h

```diff
@@ -6,6 +6,8 @@
                 #define __COMP_NPY_UNUSED __attribute__ ((__unused__))
         # elif defined(__ICC)
                 #define __COMP_NPY_UNUSED __attribute__ ((__unused__))
+        # elif defined(__clang__)
+                #define __COMP_NPY_UNUSED __attribute__ ((unused))
```

Comment:
How did you find out this was missing? What's the effect under compilers where this is not available / defined?

Reply:
The BSD thread was about warnings issued by clang, the compiler they are replacing gcc with, so I put it in for portability. I don't know if we are compiling with clang on Apple yet. The warning problem seems to be compiler dependent, even up to compiler version, and it was inconclusive if it was a compiler bug or not, or even specified by the C standards. I suspect we will need to deal with this on a case by case basis. It would be nice if we could figure out a MSVC version just for completeness. @zooba Any ideas?

Reply:
We inherited the troublesome macro from BSD when we borrowed their code for some fallback functions.